lib/base-service.ts

```diff
@@ -65,6 +66,7 @@ export class BaseService {
    */
   constructor(userOptions: UserOptions) {
     if (!(this instanceof BaseService)) {
+      logger.error('Error creating an instance of BaseService: the "new" keyword is required to create service instances');
       throw new Error(
         'the "new" keyword is required to create service instances'
```

Comment:
This is picky but the string `the "new" keyword is required to create service instances` _could_ be assigned to a variable since it is used twice here. Not a required change